api-tests/CMakeLists.txt

```diff
@@ -647,6 +648,8 @@ if(${CC312_LEGACY_DRIVER_API_ENABLED})
 endif()
 
 # Build PAL NSPE LIB
+list(APPEND PAL_SRC_C_NSPE
```

Comment:
I'm not sure this is a right approach. PAL_SRC_C_NSPE is platform/targets/xxx/target.cmake list of files and the pal_weak.c is also platform/targets file.
According to the overall project, platform/targets files must be added in the PAL_SRC_C_NSPE list from the platform/targets/xxx/target.cmake.

In my mind, adding a "platform/targets/common/nspe/target.cmake" including this file will be better, but I think will be better to highlight this problem in the PR proposal to ask what is the best approach in order to fit ARM test suite implementation rule.

Reply:
Good point.
I will push a fixup commit implementing this. And I will ask ARM to give their recommendation.